oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2Configurer.java

```diff
@@ -75,6 +79,20 @@ public void init(HttpSecurity http) throws Exception {
                 } else {
                     log.debug("OAuth resource server not configured due to missing issuer property");
                 }
+
+                if (oktaOAuth2Properties.isOpaque() && !isEmpty(resourceServerProperties.getOpaquetoken().getIntrospectionUri())) {
+                    OpaqueTokenIntrospector introspectionClient = new NimbusOpaqueTokenIntrospector(
+                        resourceServerProperties.getOpaquetoken().getIntrospectionUri(),
+                        oktaOAuth2Properties.getClientId(),
+                        oktaOAuth2Properties.getClientSecret());
```

Comment:
Tthis logic _should_ be available in the context: `context.getBean(OpaqueTokenIntrospector.class)`

Care needs to be taken here so both `http.oauth2ResourceServer().jwt()` and `http.oauth2ResourceServer().opaqueToken()` are both not called as this WILL create a problem.

It maybe possible to inject the bits we need at different level now (I haven't looked in a while)

Reply:
Should we deal with separating these flows at `runtime` or during `init`?

Reply:
addressed this.